Purchases/CodableExtensions/PurchaseOwnershipType+Extensions.swift

```diff
@@ -17,15 +17,17 @@ extension PurchaseOwnershipType: Decodable {
 
     public init(from decoder: Decoder) throws {
         let container = try decoder.singleValueContainer()
-        let purchaseOwnershipTypeString = try container.decode(String.self)
+        guard let purchaseOwnershipTypeString = try? container.decode(String.self) else {
+            throw CodableError.valueNotFound(PurchaseOwnershipType.self)
+        }
 
         switch purchaseOwnershipTypeString {
         case "PURCHASED":
             self = .purchased
         case "FAMILY_SHARED":
             self = .familyShared
         default:
-            Logger.warn("received unknown ownershipType: \(purchaseOwnershipTypeString)")
+            Logger.error(Strings.codable.unexpectedValueError(for: PurchaseOwnershipType.self))
             self = .unknown
```

Comment:
This is a special case. Here we don't throw the error, we return the `unknown` case and log the error.